### PR TITLE
Make DNS record TTL configurable by user

### PR DESCRIPTION
**Pull request by shashidharatd**
Addresses https://github.com/kubernetes-sigs/federation-v2/issues/201

/cc @font @kubernetes-sigs/federation-wg

### CODE REVIEW

**Comment by marun** [comment]
lgtm, but I'll leave it to @font to approve.

File: pkg/apis/multiclusterdns/v1alpha1/multiclusterservicednsrecord_types.go [hunk 1]
```diff
@@ -27,6 +27,8 @@ type MultiClusterServiceDNSRecordSpec struct {
 	FederationName string `json:"federationName,omitempty"`
 	// DNSSuffix is the suffix (domain) to append to DNS names
 	DNSSuffix string `json:"dnsSuffix,omitempty"`
+	// RecordTTL is the TTL for DNS records created for this Service, if omitted a default would be used
```
Comment (font) [inline pkg/apis/multiclusterdns/v1alpha1/multiclusterservicednsrecord_types.go]: Can we add the unit of time measurement for this field either as a comment or as part of the field name itself e.g. `RecordTTLSeconds`?

Also, I just wanted to point out that we use the `TTL` type in the `EndPoint` API but not here. I don't necessarily have a preference either way.

**Comment by gyliu513** [comment]
@shashidharatd it would be great to add some user guide for multicluster service DNS test this as well, so that people can try this.

File: pkg/apis/multiclusterdns/v1alpha1/multiclusterservicednsrecord_types.go (hunk 1 above)
Comment (shashidharatd) [reply pkg/apis/multiclusterdns/v1alpha1/multiclusterservicednsrecord_types.go]: Have updated the comment to specify the unit is in Seconds. Also updated the type to TTL to be in sync.

**Comment by shashidharatd** [comment]
@font, have updated this pr to address your comments. PTAL.
@gyliu513, updating the user guide is on the way. some functional part is still in progress in external-dns in https://github.com/kubernetes-incubator/external-dns/pull/657. Would update the user guide as soon as it is merged.

**Pull request merged**